### PR TITLE
Use Pipenv instead of requirements.txt to manage dependencies

### LINKED ISSUE #135: Use pip-tools
https://github.com/jazzband/pip-tools
4 minute talk explaining why: https://www.youtube.com/watch?v=LAig6s9Hkj0

### PR DESCRIPTION
**Pull request by hugobessa**
## Description

Replaces requirements.txt with Pipfile and adds configuration and documentation to start using pipenv by default in projects generated by this boilerplate.

## Motivation and Context

Resolves #135 

## Steps to reproduce (if appropriate):
- [ ] Crete a new project with the template
- [ ] Create a repository in github
- [ ] Create a project in CircleCI
- [ ] Test if CircleCI build works
- [ ] Create a new app in heroku using app.json (heroku button)
- [ ] Test if heroku build works

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires documentation updates.
- [x] I have updated the documentation accordingly.
- [x] My change requires dependencies updates.
- [x] I have updated the dependencies accordingly.

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
       "url": "heroku/nodejs"
     },
     {
-      "url": "heroku/python"
+      "url": "https://github.com/kennethreitz/heroku-buildpack-pipenv"
```
Comment (aericson) [inline app.json]: This repo does not exist. Also, heroku works with pipenv out of the box, shouldn't need a special buildpack.

File: app.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "{{project_name}}",
   "description": "{{project_name|title}} Heroku app.",
   "scripts": {
-    "postdeploy": "python manage.py migrate"
+    "postdeploy": "pipenv run python manage.py migrate"
```
Comment (aericson) [inline app.json]: Don't think you need that.

File: bin/post_compile [hunk 3]
```diff
@@ -10,11 +10,11 @@ if [ -f bin/run_collectstatic ]; then
 fi
 
 echo "-----> Running manage.py check --deploy --fail-level WARNING"
-python manage.py check --deploy --fail-level WARNING
+pipenv run python manage.py check --deploy --fail-level WARNING
```
Comment (aericson) [inline bin/post_compile]: Don't think you need that.

File: bin/post_compile [hunk 4]
```diff
@@ -10,11 +10,11 @@ if [ -f bin/run_collectstatic ]; then
...
 
 if [ -n "$AUTO_MIGRATE" ]; then
     echo "-----> Running manage.py migrate"
-    python manage.py migrate --noinput
+    pipenv run python manage.py migrate --noinput
```
Comment (aericson) [inline bin/post_compile]: Don't think you need that.

File: bin/run_collectstatic [hunk 5]
```diff
@@ -10,6 +10,6 @@ MANAGE_FILE=$(find . -maxdepth 3 -type f -name 'manage.py' | head -1)
 MANAGE_FILE=${MANAGE_FILE:2}
 
 echo "-----> Collecting static files"
-python $MANAGE_FILE collectstatic --noinput  2>&1 | sed '/^Copying/d;/^$/d;/^ /d' | indent
+pipenv run python $MANAGE_FILE collectstatic --noinput  2>&1 | sed '/^Copying/d;/^$/d;/^ /d' | indent
```
Comment (aericson) [inline bin/run_collectstatic]: Don't think you need that.

File: package.json [hunk 6]
```diff
@@ -42,7 +42,7 @@
     "script-loader": "^0.7.0",
     "style-loader": "^0.18.2",
     "url-loader": "^0.5.9",
-    "webpack": "^3.8.1",
+    "webpack": "^2.6.1",
```
Comment (aericson) [inline package.json]: Why downgrading?

File: package.json [hunk 7]
```diff
@@ -23,7 +23,7 @@
     "classnames": "^2.2.5",
     "css-loader": "^0.28.4",
     "expose-loader": "^0.7.3",
-    "extract-text-webpack-plugin": "^3.0.2",
+    "extract-text-webpack-plugin": "^2.1.2",
```
Comment (aericson) [inline package.json]: Why downgrading?

File: Pipfile [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+[packages]
+
+Django = "==1.11.8"
+celery = {extras = ["redis"]}
+django-model-utils = "*"
+django-webpack-loader = "*"
+django-js-reverse = "*"
+django-import-export = "*"
+python-decouple = "<=4.0"
```
Comment (aericson) [inline Pipfile]: I've always wondered, why are we doing this? Either we do that for all libs or none at all.

File: Pipfile [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+django-naomi = "*"
+fixmydjango = "*"
+model-mommy = "*"
+pre_commit = "*"
+
+
+
+[packages]
+
+Django = "==1.11.8"
```
Comment (aericson) [inline Pipfile]: Why locking django version to a minor version?

File: package.json (hunk 7 above)
Comment (hugobessa) [reply package.json]: this was probably a merge mistake

File: package.json (hunk 6 above)
Comment (hugobessa) [reply package.json]: this was probably a merge mistake

File: Pipfile (hunk 8 above)
Comment (hugobessa) [reply Pipfile]: Better to open another issue to discuss that since the problem is not related to pipenv

File: README.md [hunk 10]
```diff
@@ -56,9 +58,10 @@ After completing ALL of the above, remove this `Project bootstrap` section from
 - Setup [editorconfig](http://editorconfig.org/), [prospector](https://prospector.landscape.io/en/master/) and [ESLint](http://eslint.org/) in the text editor you will use to develop.
 
 ### Running the project
-- `pip install -r requirements.txt`
+- `pipenv install`
```
Comment (fjsj) [inline README.md]: shouldn't be `pipenv install --dev`?

**Comment by fjsj** [comment]
LGTM

**Comment by hugobessa** [comment]
@aericson @chocoelho I've frozen the version of astroid (dependency of pylint, https://github.com/PyCQA/astroid/) to make prospector work again. The error happened because both pylint and astroid updated yesterday, probably with buggy changes. I guess the linters/linters wrappers discussion should happen in a separate issue to address the problem without interference caused by other unrelated features.

File: Pipfile (hunk 8 above)
Comment (hugobessa) [reply Pipfile]: discussion about that here #164

**Comment by hugobessa** [comment]
Discussion about prospector replacement here #165

**Comment by chocoelho** [comment]
Awesome @hugobessa. I agree with you that replacement should be discussed in another issue so we can keep the scope of this PR to Pipenv only.

**Pull request merged**